File: ClientJava/network/src/main/java/network/server/ServerUDP.java
package network.server;

import network.message.PacketMessage;
import network.share.DataListener;
import network.share.ListenerState;
import network.udp.NetworkManagerUDP;
import network.udp.ProtocolEventsUDP;

import java.io.IOException;
import java.net.SocketException;
import java.net.UnknownHostException;

public class ServerUDP implements Runnable {

    NetworkManagerUDP manager;

    public ServerUDP(ListenerState listener) throws SocketException, UnknownHostException {
        manager = new NetworkManagerUDP(listener);
    }

    public <T> void OnEvent(Class<T> co, ProtocolEventsUDP<T> proto, DataListener<T> listener){
        manager.evtManager.OnEvent(co,proto,listener);
    }

    public void send(PacketMessage<?> obj) throws IOException {
        manager.sendMessage(obj);
    }

    @Override
    public void run() {
        try {
            manager.startListening();
        } catch (IOException e) {
            e.printStackTrace();
            Thread.currentThread().interrupt();
        }
    }
}

File: ProjectTest/share/Network/Server/ServerUDP.cs
using share;
using Share.Network.Message;
using Share.Network.Message.obj;
using Share.Network.NetworkManager;
using Share.Network.Protocol;
using Share.Network.Protocol.UDP;
using System;
using System.Collections.Generic;
using System.Net;
using System.Reflection.Metadata.Ecma335;
using System.Text;
using System.Threading;

namespace Share.Network.Server
{
    public class ServerUDP
    {

        NetworkManagerUDP manager;

        public bool IsRunning { get => manager.running;  }

        public ServerUDP()
        {
            manager = new NetworkManagerUDP();
        }

        public void OnEvent<T>(ProtocolEvents<T> proto, ProtocolUDP<T>.OnReceiveEvent callback)
        {
            manager.evtNetManager.OnEvent(proto, callback);
        }

        public void Send<T>(PacketMessage<T> data, EndPoint ep)
        {
            manager.Send(data, ep);
        }

        public void Run()
        {
            manager.StartListening();
        }

    }
}
